evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +123,44 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectByTypeorTag(ColumnSelector):
+    """
+    Selects columns by specified Woodwork logical type or semantic tag in input data.
+
+    Arguments:
+        columns (list(string), list(ww.LogicalType)): List of datatypes, used to determine which columns to select.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Select Columns By Type or Tag Transformer"
+    hyperparameter_ranges = {}
+    """{}"""
+    needs_fitting = False
+
+    def _check_input_for_columns(self, X):
+        """
+        This check is not required, since ww.select does not throw an error if the requested types are not in the dataset.
+        """
+        return
+
+    def _modify_columns(self, cols, X, y=None):
+        columns = X.ww.select(cols)
+        if len(cols) > 0 and columns.empty:
+            warnings.warn(
+                "No columns of the selected type(s) were found in the input data. SelectByTypeorTag will return an empty DataFrame"
+            )
+        return X.ww.select(cols)
+
+    def transform(self, X, y=None):
+        """Transforms data X by selecting columns.
+
+        Arguments:
+            X (pd.DataFrame): Data to transform.
+            y (pd.Series, optional): Targets.
+
+        Returns:
+            pd.DataFrame: Transformed X.
+        """
+        return super().transform(X, y)
```

Comment:
If all the `transform` method for this class is doing is calling `super`, please delete it!

Reply:
The downside is that we lose the docstring! But since this docstring isn't very specific to this class, that's ok I think